src/wayland-eglsurface.c

```diff
@@ -2895,6 +2895,7 @@ EGLSurface wlEglCreatePbufferSurfaceHook(EGLDisplay dpy,
     surface->refCount = 1;
     surface->isDestroyed = EGL_FALSE;
     wl_list_init(&surface->oldCtxList);
+    wl_list_init(&surface->ctx.streamImages);
```

Comment:
This seems like a correct addition, but are there other fields that are missing initialization here? E.g., surface->ctx.streamImageMutex?

Would it be a better fix to have a common surface initializer helper function that covers all the surface-type-agnostic intialization of a surface structure?

Reply:
Thanks, added this.